src/svc_rqst.c

```diff
@@ -322,17 +326,19 @@ consume_ev_sig_nb(int fd)
 static inline void
 svc_rqst_release(struct svc_rqst_rec *sr_rec /* LOCKED => UNLOCKED */)
 {
-	uint32_t refcnt;
-
-	refcnt = --(sr_rec->refcnt);
-
-	mutex_unlock(&sr_rec->mtx);
-
-	if (refcnt)
+	if (--(sr_rec->refcnt)) {
+		mutex_unlock(&sr_rec->mtx);
 		return;
+	}
 
 	/* assert sr_rec DESTROYED */
 	svc_rqst_set.srr[sr_rec->id_k] = NULL;
```

Comment:
This needs to be protected by svc_rqst_set.mtx, but taking it here inverts lock order with svc_rqst_lookup_chan()